### ISSUE TITLE
Functions to randomize and slice PAGES collection in filter_functions.php

### ISSUE THREAD

**Issue by dimayakovlev** [GetSimpleCMS/GetSimpleCMS#1195]
These built in functions may be useful for work with filtered PAGES collections.

**Comment by dimayakovlev** [comment]
I use this function to shuffle pages collections.
```
function shuffle_assoc($list) { 
  if (!is_array($list)) return $list; 

  $keys = array_keys($list); 
  shuffle($keys); 
  $random = array(); 
  foreach ($keys as $key) 
    $random[$key] = $list[$key]; 

  return $random; 
}
```

**Comment by dimayakovlev** [comment]
Also very good to have ability to sort PAGES collection by PAGES fields.

**Comment by tablatronix** [comment]
What would you use randomized pages for?

**Comment by dimayakovlev** [comment]
For tasks where need to use random pages order.
For example, I want to show five random pages with special keywords and don't want to write my own functions:
```php
$collection = array_keys(getPages('filterTags', array('key1', 'key2', 'key3'), false, true, false));
shuffle($collection);
$collection = array_slice($collection, 0, 5);
foreach ($collection as $item) {
  echo '<h4>'.returnPageField($item, 'title').'</h4><p>'.returnPageField($item, 'summary').'</p>';
}
```
But I think better to work with pages collection than array of slugs. And if there is function to sort collection by page field I can sort shuffled and sliced collection before creating output.

**Comment by tablatronix** [comment]
yeah it should be fairly easy to chain functions for this stuff now, eventually it can be actually chained when it is a class.

Since collections are just arrays you can just array shuffle no?

**Comment by dimayakovlev** [comment]
Collections are associative arrays, so need special function to shuffle them. I posted example of that function, but to use it must be placed in plugin or in theme. That's why I'm asking about built in function.

**Comment by tablatronix** [comment]
too bad it is not key safe, does it replace the keys or does it just shuffle values around with the same keys ?

**Comment by dimayakovlev** [comment]
I use it and have no problems. Function returns shuffled array with preserved key => value pairs.

**Comment by tablatronix** [comment]
I was referring to `shuffle`

**Comment by dimayakovlev** [comment]
Ahh, `shuffle` will delete keys.
